js/main.js

```diff
@@ -1,281 +1,422 @@
-    var randomItem = getRandomNumber(0, newList.length - 1);
+function getOnlyNumberFromStr(strWithNumbers) {
+  var needSymbols = /[0-9/.]+/;
+  return strWithNumbers.match(needSymbols);
+}
 
-    newList.splice(randomItem, 1);
+function activeMapPins() {
+  var mapPinTemplate = document.querySelector('#pin')
+    .content
+    .querySelector('.map__pin');
+
+  var MAP_WIDTH = 1200;
+  var PIN_HEIGHT = 70;
+  var PIN_WIDTH = 50;
+
+  var ADS_NUMBER = 8;
+  var MAX_MAP_Y = 630;
+  var MIN_MAP_Y = 130;
+  var MAX_MAP_X = MAP_WIDTH - PIN_WIDTH;
+  var MIN_PRICE = 200;
+  var MAX_PRICE = 2500;
+
+  function createAd() {
+    for (var i = 0; i < ADS_NUMBER; i++) {
+      var userAvatar = 'img/avatars/user0' + (i + 1) + '.png';
+
+      var apartmentPrice = getRandomNumber(MIN_PRICE, MAX_PRICE);
+      var apartmentType = getRandomItemFromArray(apartmentsTyps);
+      var apartmentRooms = getRandomNumber(1, 5);
+      var apartmentGuests = getRandomNumber(0, 8);
+      var apartmentCheckin = getRandomItemFromArray(timesCheckin);
+      var apartmentCheckout = getRandomItemFromArray(timesCheckout);
+      var apartmentFeatures = getRandomList(apartmentsFeatures);
+      var apartmentPhotos = getRandomList(apartmentsPhotos);
+
+      var positionX = getRandomNumber(0, MAX_MAP_X);
+      var positionY = getRandomNumber(MIN_MAP_Y, MAX_MAP_Y);
+
+      var apartmentTittle = apartmentType + getRandomList(apartmentsTitleSale);
+      var apartmentDescription = apartmentTittle + '. ' + 'Количество комнат: ' + apartmentRooms + '. ' +
+        'Максимальное количество человек: ' + apartmentGuests + '. ' + 'Подходит как туристам, так и бизнесменам. Квартира полностью укомплектована и недавно отремонтирована.';
+
+      usersAds[i] = {
+        author: {
+          avatar: userAvatar,
+        },
+        offer: {
+          tittle: apartmentTittle,
+          address: positionX,
+          price: apartmentPrice,
+          type: apartmentType,
+          rooms: apartmentRooms,
+          guests: apartmentGuests,
+          checkin: apartmentCheckin,
+          checkout: apartmentCheckout,
+          features: apartmentFeatures,
+          description: apartmentDescription,
+          photos: apartmentPhotos,
+        },
+        location: {
+          x: positionX,
+          y: positionY,
+        },
+      };
+    }
   }
 
-  return newList;
-}
+  function getRandomNumber(minNumber, maxNumber) {
+    return Math.floor(Math.random() * (maxNumber - minNumber + 1)) + minNumber;
+  }
 
-createAd();
+  function getRandomItemFromArray(items) {
+    return items[getRandomNumber(0, items.length - 1)];
+  }
 
-function renderMapPin(userAd) {
-  var mapPinElement = mapPinTemplate.cloneNode(true);
-  var mapPinStyle = 'left: ' + (userAd['location']['x'] - PIN_WIDTH / 2) + 'px; ' +
-    'top: ' + (userAd['location']['y'] - PIN_HEIGHT) + 'px;';
+  function getRandomList(items) {
+    var newList = items.slice(0);
+    var excessValue = getRandomNumber(0, items.length);
```

Comment:
```suggestion
    var excessValue = getRandomNumber(0, items.length-1);
```
И тогда следующий `if` можно будет убрать